packages/frontend/src/components/Setup/FormStep/index.test.jsx

```diff
@@ -0,0 +1,66 @@
+import { setupTests } from 'helpers/tests';
+import FormStep from '.';
+import { routes } from '../../../utils';
+import { baseState } from 'contexts';
+import { produce } from 'immer';
+
+describe('FormStep component', () => {
+  it('renders the correct label and form element at step one', () => {
+    const { label, placeholder, buttonText, buttonColorClass, FormElement } =
+      routes[0].children[0].elementProps;
+    const { name, schema } = routes[0].children[0];
+
+    const props = {
+      label,
+      placeholder,
+      buttonText,
+      buttonColorClass,
+      name,
+      schema,
+      FormElement,
+    };
+    const state = { ...baseState };
+
+    const { screen } = setupTests(FormStep, { props, state });
+
+    const stepOneComponent = screen.getByText("Enter the first player's name");
+
+    expect(stepOneComponent).toBeInTheDocument();
+
+    expect(screen.queryByText("Enter the second player's name")).toBeNull();
+
+    const input = document.querySelector('input[type="name"]');
```

Comment:
I think this could use `getByLabelText` instead of `document.querySelector`.

Reply:
Looking for clarification on this. Do you mean not to check if the correct form element is rendered? Like we are not checking if an input or a slider is rendered, just that the button text, label text, etc. appears correctly according to the passed props?

Reply:
@perennialAutodidact I've updated this PR. If I'm reading your note correctly, I think the label text ( maybe its actually header text because its not in a label tag but an h3 ) is the only thing we want to test at each step? The other props are for components, namely the form element and button component, which we are not working with in this test.

Reply:
`getByLabelText` targets the text within the `<label>` element associated with a particular input. One of the philosophies of React Testing Library is that the tests should be relative to what the user would see rendered in the dom. The user doesn't care if `input[type="name"]` is there. In fact, that input could exist and be invisible. The user would look for the label text and then assume that the input was associated with it. This is how the test should be written as well, according to RTL.

Reply:
@perennialAutodidact The updated version of the test doesn't have this in it. It only checks for the correct text content at each step.